src/Combinatorics/EnumerativeCombinatorics/multipartitions.jl

```diff
@@ -0,0 +1,168 @@
+################################################################################
+# Multipartitions.
+#
+# Copyright (C) 2020 Ulrich Thiel, ulthiel.com/math
+################################################################################
+
+
+
+
+"""
+    multipartition(mp::Vector{Vector{T}}) where T <: IntegerUnion
+
+Return the multipartition given by the vector `mp` of integer sequences `mp`
+(which are interpreted as integer partitions) as an object of type
+`Multipartition{T}`.
+
+The element type `T` may be optionally specified, see also the examples below.
+
+# Examples
+```julia-repl
+julia> P=Multipartition( [[2,1], [], [3,2,1]] )
+Partition{Int64}[[2, 1], [], [3, 2, 1]]
+julia> sum(P)
+9
+julia> P[2]
+Int64[]
+julia> P=Multipartition( Vector{Int8}[[2,1], [], [3,2,1]] )
+Partition{Int8}[[2, 1], [], [3, 2, 1]]
+```
+
+# References
+1. Wikipedia, [Multipartition](https://en.wikipedia.org/wiki/Multipartition)
+"""
+function multipartition(mp::Vector{Vector{T}}) where T <: IntegerUnion
+  return Multipartition([partition(p) for p in mp])
+end
+
+# This is only called when the empty array is part of mp (because then it's
+# "Any" type and not of Integer type).
+function multipartition(mp::Vector{Vector{Any}})
+  return Multipartition([partition(p) for p in mp])
+end
+
+
+function Base.show(io::IO, ::MIME"text/plain", MP::Multipartition)
+  print(io, MP.mp)
+end
+
+function Base.size(MP::Multipartition)
+  return size(MP.mp)
+end
+
+function Base.length(MP::Multipartition)
+  return length(MP.mp)
+end
+
+function Base.getindex(MP::Multipartition, i::Int)
+  return getindex(MP.mp,i)
+end
+
+
+"""
+    sum(P::Multipartition{T}) where T<:Integer
+
+If P is a multipartition of the integer n, this function returns n.
+"""
+function Base.sum(P::Multipartition{T}) where T<:Integer
+  s = zero(T)
+  for i=1:length(P)
+    s += sum(P[i])
+  end
+  return s
+end
+
+
+"""
+    multipartitions(n::T, r::Integer) where T<:Integer
```

Comment:
```suggestion
    multipartitions(n::T, r:: IntegerUnion) where T<:IntegerUnion
```
or even
```suggestion
    multipartitions(n::IntegerUnion, r::IntegerUnion)
```
I'd only use the version with `T` if the docstring is changed to mention it (e.g. to describe the type of the return value)